test/integration/config-validation/test/index.test.ts

```diff
@@ -81,6 +81,33 @@ describe('next.config.js validation', () => {
           '"env.QUX" is missing'
         )
       })
+
+      it('should allow for empty topLevelImportPaths and meaninglessFileNames arrays (compiler.styledComponents)', async () => {
+        const configContent = `
+        module.exports = {
+          compiler: {
+            styledComponents: {
+              topLevelImportPaths: [],
+              meaninglessFileNames: []
+            }
+          }
+        }`
+
+        await fs.writeFile(nextConfigPath, configContent)
+        const result = await nextBuild(path.join(__dirname, '../'), undefined, {
+          stderr: true,
+          stdout: true,
+        })
+
+        await fs.remove(nextConfigPath)
+
+        expect(result.stdout + result.stderr).not.toContain(
+          'Array must contain at least 1 element(s) at "compiler.styledComponents.topLevelImportPaths'
+        )
+        expect(result.stdout + result.stderr).not.toContain(
+          'Array must contain at least 1 element(s) at "compiler.styledComponents.meaninglessFilenames'
+        )
```

Comment:
Using `.not` here may be unreliable as test may pass on any future stdout change or unexpected omits.

Or at least we need to confirm that current behavior does contain those messages.

How about we just have an e2e test for valid configs?

Reply:
@devjiwonchoi We do test for valid configs in this array → https://github.com/vercel/next.js/pull/69230/files#diff-24bf7bb1b07277e23487cbfba47b880a643a54c878373a45725b91c48c04a2fdL11, so I could even simplify this PR by adding it that array instead.

What do you think of the below?

Something like:
```
 {
          name: 'invalid config array lengths',
          configContent: `
        module.exports = {
          pageExtensions: []
        }  
      `,
          outputs: [
            `Array must contain at least 1 element(s) at "pageExtensions"`,
          ],
        },
   ```

Reply:
Can we use that to check after the fix as well?

Reply:
> Can we use that to check after the fix as well?

https://github.com/vercel/next.js/pull/69230/files#diff-24bf7bb1b07277e23487cbfba47b880a643a54c878373a45725b91c48c04a2fdL43

This just checks if the configs are valid as well